## Quiz 4

Answer the questions in the box provided. You need to run the code and show the output for the questions which require coding or type your answer inside the box for the fill-in-the-blank questions. This is an open-book quiz. You are allowed to consult any local or online resources, except for ChatGPT. Any misconduct or cheating will result in a score of 0. Good luck!

<font size = 3> For all questions except 1-1 and 1-3, follow these steps:
* Write code to solve the question in the first box (the box labeled with the phrase #Write your code here: ).
* In the following box(es), provide the correct answer in the blank ..... <br>

<font size = 3> In order to receive a point for a question, both boxes must contain correct answers.

## Part 1

1-1 Use Pandas to read the `starbucks` dataset from `starbucks.csv` into a DataFrame.

In [1]:
import pandas as pd
pd.read_csv('starbucks.csv')


Beverage_category                                      Beverage  \
0                        Coffee                                 Brewed Coffee   
1                        Coffee                                 Brewed Coffee   
2                        Coffee                                 Brewed Coffee   
3                        Coffee                                 Brewed Coffee   
4       Classic Espresso Drinks                                   Caffè Latte   
..                          ...                                           ...   
237  Frappuccino® Blended Crème  Strawberries & Crème (Without Whipped Cream)   
238  Frappuccino® Blended Crème          Vanilla Bean (Without Whipped Cream)   
239  Frappuccino® Blended Crème          Vanilla Bean (Without Whipped Cream)   
240  Frappuccino® Blended Crème          Vanilla Bean (Without Whipped Cream)   
241  Frappuccino® Blended Crème          Vanilla Bean (Without Whipped Cream)   

          Beverage_prep  Calories  Total Fat (g)  Trans Fat (g)   \
0                 Short         3            0.1             0.0   
1                  Tall         4            0.1             0.0   
2                Grande         5            0.1             0.0   
3                 Venti         5            0.1             0.0   
4     Short Nonfat Milk        70            0.1             0.1   
..                  ...       ...            ...             ...   
237             Soymilk       320            3 2             0.4   
238    Tall Nonfat Milk       170            0.1             0.1   
239          Whole Milk       200            3.5             2.0   
240             Soymilk       180            1.5             0.2   
241  Grande Nonfat Milk       240            0.1             0.1   

     Saturated Fat (g)   Sodium (mg)   Total Carbohydrates (g)   \
0                  0.0             0                          5   
1                  0.0             0                         10   
2                  0.0             0                         10   
3                  0.0             0                         10   
4                  0.0             5                         75   
..                 ...           ...                        ...   
237                0.0             0                        250   
238                0.0             0                        160   
239                0.1            10                        160   
240                0.0             0                        160   
241                0.0             5                        230   

     Cholesterol (mg)   Dietary Fibre (g)   Sugars (g)   Protein (g)   \
0                   0                   0            0            0.3   
1                   0                   0            0            0.5   
2                   0                   0            0            1.0   
3                   0                   0            0            1.0   
4                  10                   0            9            6.0   
..                ...                 ...          ...            ...   
237                67                   1           64            5.0   
238                39                   0           38            4.0   
239                39                   0           38            3.0   
240                37                   1           35            3.0   
241                56                   0           55            5.0   

    Vitamin A (% DV)  Vitamin C (% DV)  Calcium (% DV)  Iron (% DV)   \
0                  0%               0%               0%           0%   
1                  0%               0%               0%           0%   
2                  0%               0%               0%           0%   
3                  0%               0%               2%           0%   
4                 10%               0%              20%           0%   
..                ...              ...              ...          ...   
237                6%               8%

1-2 Use Pandas to figure out how many columns and rows the DF has.

In [42]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF.shape

(242, 18)

number of columns is "18"

number of rows is "242"

1-3 Write a code that inspects the first 19 rows in the first box below and a code that inspects the last 5 rows in the second box.

In [ ]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF.head(19)

In [ ]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF.tail(5)

1-4 Which columns (if any) have missing values?

The columns which contain missing values are .....

1-5 How many floats columns are in the DataFrame?

In [39]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF.count()

Beverage_category            242
Beverage                     242
Beverage_prep                242
Calories                     242
 Total Fat (g)               242
Trans Fat (g)                242
Saturated Fat (g)            242
 Sodium (mg)                 242
 Total Carbohydrates (g)     242
Cholesterol (mg)             242
 Dietary Fibre (g)           242
 Sugars (g)                  242
 Protein (g)                 242
Vitamin A (% DV)             242
Vitamin C (% DV)             242
 Calcium (% DV)              242
Iron (% DV)                  242
Caffeine (mg)                241
dtype: int64

The number of floats columns is "18"

## Part 2

**Use `starbucks.csv` to answer the following questions:**

2-1 Find the highest Calories in the DataFrame.

In [30]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF["Calories"].max()

510

The highest calories in the DataFrame is "510"

2-2 What is the average calories, rounded to one decimal digit, of the last 56 rows?

In [26]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF["Calories"].tail(56).mean()

249.82142857142858

The average calories of the last 56 rows is "249.82142857142858 or 250"

2-3 What is the sum of all the values in the `Total Carbohydrates` column?

In [24]:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF[" Total Carbohydrates (g) "].sum()

31190

The sum of the values in the `Total Carbohydrates` column is "31190"

2-4 How many different beverage categories are in the dataset?

In [23]:
#Write your code here:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF["Beverage_category"].nunique()

9

The dataframe has "9" beverage categories.

2-5 Which beverage has the most transactions? How many times does it appear in the dataset?

In [21]:
#Write your code here:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF["Beverage"].value_counts().idxmax()


'Tazo® Full-Leaf Red Tea Latte (Vanilla Rooibos)'

In [22]:
#Write your code here:
import pandas as pd
pd.read_csv('starbucks.csv')
DF = pd.read_csv('starbucks.csv')
DF["Beverage"].value_counts().max()

12

The name of the beverage which has the most transactions is "Tazo® Full-Leaf Red Tea Latte (Vanilla Rooibos)"

It appears in the dataset "12" times.